|(a_i−1, a_i]|n_i|
|:-:|:-:|
|0-24|1|
|24-48|2|
|48-72|4|
|72-96|6|
|96-120|12|
|120-144|16|
|144-168|6|
|168-192|2|
|192-216|1|

In [160]:
# (a_i−1, a_i] = interval
alpha = 0.01
H_0 = {"Total population has a normal distribution."}

In [161]:
interval = [[0, 24], [24, 48], [48, 72], [72, 96], [96, 120], [120, 144], [144, 168], [168, 192], [192, 216]]
n_i = [1, 2, 4, 6, 12, 16, 6, 2, 1]

In [162]:
# mean of intervals
x_i = []
for i in interval:
    x_i.append((i[0]+i[1])//2)
x_i

[12, 36, 60, 84, 108, 132, 156, 180, 204]

In [163]:
# n and x_bar
n = sum(n_i)
a = []
for i1, i2 in zip(x_i, n_i):
    a.append(i1*i2)
A = sum(a)
x_bar = (1/n)*A
x_bar

114.72

In [164]:
# Dispersion
b = []
for i1, i2 in zip(x_i, n_i):
    b.append((i1**2)*i2)
B = sum(b)
D_bar = B/n - x_bar**2
D_bar

1452.441600000002

In [165]:
# deviation
sigma_bar = D_bar**(1/2)
sigma_bar

38.1109118232561

In [166]:
# a_i-1, a_i to z_i-1, z_i
# z_i-1 - z_i = new_interval
# z_i = (a_i - x_bar) / sigma_bar
# z_i-1 = (a_i-1 - x_bar) / sigma_bar
new_interval = interval
z_0 = float("-inf")
z_m = float("inf")
for i in new_interval:
    i[0] = round((i[0]-x_bar)/sigma_bar, 2)
    i[1] = round((i[1]-x_bar)/sigma_bar, 2)
new_interval[0][0] = z_0
new_interval[8][1] = z_m
new_interval

[[-inf, -2.38],
 [-2.38, -1.75],
 [-1.75, -1.12],
 [-1.12, -0.49],
 [-0.49, 0.14],
 [0.14, 0.77],
 [0.77, 1.4],
 [1.4, 2.03],
 [2.03, inf]]

**Filling the PHI value in accordance with table of Laplace transforms:**

|i|(z_i-1, z_i]|PHI(z_i-1)|PHI(z_i)|
|:-:|:-:|:-:|:-:|
|1|(-inf, -2.38]|-0.5|-0.4913|
|2|(-2.38, -1.75]|-0.4913|-0.4599|
|3|(-1.75, -1.12]|-0.4599|-0.3686|
|4|(-1.12, -0.49]|-0.3686|-0.1879|
|5|(-0.49, 0.14]|-0.1879|0.0557|
|6|(0.14, 0.77]|0.0557|0.2794|
|7|(0.77, 1.4]|0.2794|0.4192|
|8|(1.4, 2.03]|0.4192|0.4788
|9|(2.03, inf)|0.4788|0.5|

In [167]:
PHI_z_i_minus_1 = [-0.5, -0.4913, -0.4599, -0.3686, -0.1879, 0.0557, 0.2794, 0.4192, 0.4788]
PHI_z_i = [-0.4913, -0.4599, -0.3686, -0.1879, 0.0557, 0.2794, 0.4192,  0.4788, 0.5]

In [168]:
# p_i = PHI(z_i) - PHI(z_i-1)
p_i = []
for i1, i2 in zip(PHI_z_i, PHI_z_i_minus_1):
    p_i.append(round(i1 - i2, 4))
p_i

[0.0087, 0.0314, 0.0913, 0.1807, 0.2436, 0.2237, 0.1398, 0.0596, 0.0212]

In [169]:
# n_i_bar = n * p_i
n_i_bar = []
for i in p_i:
    n_i_bar.append(round(i*n, 3))
n_i_bar

[0.435, 1.57, 4.565, 9.035, 12.18, 11.185, 6.99, 2.98, 1.06]

In [170]:
# K_empirical = sum((n_i-n_i_bar)**2/n_i_bar)
c = []
for i1, i2 in zip(n_i, n_i_bar):
    c.append((i1-i2)**2/i2)
K_empirical = round(sum(c), 3)
K_empirical

4.482

In [171]:
# r = m - s - 1
m = len(interval)
s = 2
r = m - s - 1
r

6

In [172]:
# In accordance with CHI-square Distribution Table: k_critical = CHI_square(alpha, r)
k_critical = 16.8
H_0 = K_empirical < k_critical
H_0

True

**H0 returns True after calculations performed. Hypothesis H0 is accepted.**